standalone-metastore/src/main/thrift/hive_metastore.thrift

```diff
@@ -861,6 +861,18 @@ struct AbortTxnsRequest {
 struct CommitTxnRequest {
     1: required i64 txnid,
     2: optional string replPolicy,
+    // Information related to write operations done in this transaction.
+    3: optional list<WriteEventInfo> writeEventInfos,
+}
+
+struct WriteEventInfo {
+    1: required i64    writeId,
+    2: required string database,
+    3: required string table,
+    4: required string partition,
```

Comment:
Will partition value present for non-partitioned table write?
Also, why files and tableObj are optional?

Reply:
if table and partition are optional, the no point in setting files are required as for read txns, the files list is empty.

Reply:
WriteEventInfo is set only for write txns

Reply:
ok. So, table and files are required fields where as partition is optional. Is that right?

Reply:
yes